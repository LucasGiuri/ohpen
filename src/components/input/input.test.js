import Input from './input';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { render, fireEvent, screen } from '@testing-library/react';

describe('Input', () => {
  const dataTestPrefix = 'dataTestPrefix';
  const defaultValue = 'defaultValue';
  const options = ['', 'defaultValueA', 'defaultValueB', 'defaultValue'];
  const onChange = jest.fn();

  describe('Snapshots', () => {
    it('default snapshot render', () => {
      const tree = renderer.create(<Input />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  
    it('with custom dataTestPrefix snapshot render', () => {
      const tree = renderer.create(<Input dataTestPrefix={dataTestPrefix} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  
    it('with custom defaultValue snapshot render', () => {
      const tree = renderer.create(<Input defaultValue={defaultValue} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  
    it('with type text snapshot render', () => {
      const tree = renderer.create(<Input type='text' />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('with type text and defaultValue snapshot render', () => {
      const tree = renderer.create(<Input type='text' defaultValue={defaultValue} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  
    it('with type dropdown snapshot render', () => {
      const tree = renderer.create(<Input type='dropdown' options={options} />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('with type dropdown and defaultValue snapshot render', () => {
      const tree = renderer.create(<Input type='dropdown' options={options} defaultValue={defaultValue} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
