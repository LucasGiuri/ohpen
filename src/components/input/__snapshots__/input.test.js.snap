// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Snapshots default snapshot render 1`] = `
<div
  className="sc-gsTEea cNPdrl"
  data-test="input-input-container"
>
  <label
    className="sc-eCstlR egbkxO"
  />
  <div
    className="sc-bdfBQB jINbmp"
    height=""
    width=""
  />
  <input
    className="sc-dlfnuX epJnGT"
    data-test="input-input"
    onChange={[Function]}
    value=""
  />
</div>
`;

exports[`Input Snapshots with custom dataTestPrefix snapshot render 1`] = `
<div
  className="sc-gsTEea cNPdrl"
  data-test="dataTestPrefix-input-container"
>
  <label
    className="sc-eCstlR egbkxO"
  />
  <div
    className="sc-bdfBQB jINbmp"
    height=""
    width=""
  />
  <input
    className="sc-dlfnuX epJnGT"
    data-test="dataTestPrefix-input"
    onChange={[Function]}
    value=""
  />
</div>
`;

exports[`Input Snapshots with custom defaultValue snapshot render 1`] = `
<div
  className="sc-gsTEea cNPdrl"
  data-test="input-input-container"
>
  <label
    className="sc-eCstlR egbkxO"
  />
  <div
    className="sc-bdfBQB jINbmp"
    height=""
    width=""
  />
  <input
    className="sc-dlfnuX epJnGT"
    data-test="input-input"
    onChange={[Function]}
    value="defaultValue"
  />
</div>
`;

exports[`Input Snapshots with type dropdown and defaultValue snapshot render 1`] = `
<div
  className="sc-gsTEea cNPdrl"
  data-test="input-input-container"
>
  <label
    className="sc-eCstlR egbkxO"
  />
  <div
    className="sc-bdfBQB jINbmp"
    height=""
    width=""
  />
  <select
    className="sc-hKgJUU cWPyzQ"
    data-test="input-input-select"
    onChange={[Function]}
    value="defaultValue"
  >
    <option>
      
    </option>
    <option>
      defaultValueA
    </option>
    <option>
      defaultValueB
    </option>
    <option>
      defaultValue
    </option>
  </select>
</div>
`;

exports[`Input Snapshots with type dropdown snapshot render 1`] = `
<div
  className="sc-gsTEea cNPdrl"
  data-test="input-input-container"
>
  <label
    className="sc-eCstlR egbkxO"
  />
  <div
    className="sc-bdfBQB jINbmp"
    height=""
    width=""
  />
  <select
    className="sc-hKgJUU cWPyzQ"
    data-test="input-input-select"
    onChange={[Function]}
    value=""
  >
    <option>
      
    </option>
    <option>
      defaultValueA
    </option>
    <option>
      defaultValueB
    </option>
    <option>
      defaultValue
    </option>
  </select>
</div>
`;

exports[`Input Snapshots with type text and defaultValue snapshot render 1`] = `
<div
  className="sc-gsTEea cNPdrl"
  data-test="input-input-container"
>
  <label
    className="sc-eCstlR egbkxO"
  />
  <div
    className="sc-bdfBQB jINbmp"
    height=""
    width=""
  />
  <input
    className="sc-dlfnuX epJnGT"
    data-test="input-input"
    onChange={[Function]}
    value="defaultValue"
  />
</div>
`;

exports[`Input Snapshots with type text snapshot render 1`] = `
<div
  className="sc-gsTEea cNPdrl"
  data-test="input-input-container"
>
  <label
    className="sc-eCstlR egbkxO"
  />
  <div
    className="sc-bdfBQB jINbmp"
    height=""
    width=""
  />
  <input
    className="sc-dlfnuX epJnGT"
    data-test="input-input"
    onChange={[Function]}
    value=""
  />
</div>
`;
